tools/esdlcmd/esdlcmd_manifest.cpp

```diff
@@ -0,0 +1,1409 @@
+            EndTag etag;
+            // Track the node level nesting of nodes handled in this function only
+            int level = 1;
+
+            // To preserve order, the expected format for scripts is:
+            // <Scripts>
+            //     <![CDATA[<Scripts>...actual scripts...</Scripts>]]>
+            // </Scripts>
+
+            pushNamespaces(xpp, &root);
+            // If an included file defines any namespaces in its root <Scripts> node
+            // then cache those ns definitions in this frame
+            NamespaceFrame nsForEnclosedScripts(newNamespaces.top());
+
+            appendCompleteStartTag(MANIFEST_TAG_SCRIPTS);
+            appendStartCDATA();
+            appendCompleteStartTag(MANIFEST_TAG_SCRIPTS);
+            int scriptsNsInsertOffset = binding.length() - 1;
+
+            do
+            {
+                type = xpp->next();
+                switch (type)
+                {
+                    case XmlPullParser::START_TAG:
+                        xpp->readStartTag(stag);
+                        if (isManifestNodeNamed(stag, MANIFEST_TAG_INCLUDE))
+                            processInclude(xpp, stag, INCLUDE_SCRIPT, &nsForEnclosedScripts);
+                        else
+                        {
+                            appendStartTag(xpp, stag);
+                            level++;
+                        }
+                        break;
+
+                    case XmlPullParser::CONTENT:
+                        appendNonWhitespaceContent(xpp);
+                        break;
+
+                    case XmlPullParser::END_TAG:
+                        xpp->readEndTag(etag);
+
+                        if (isRootEndTagNamed(etag, "Scripts", level))
+                        {
+                            // Insert cached included namespaces to the cdata-enclosed <Scripts> start tag
+                            // then append end tags for those start tags explicitly appended.
+                            if (nsForEnclosedScripts.size() > 0)
+                            {
+                                StringBuffer nsbuf;
+                                for (std::string ns : nsForEnclosedScripts)
+                                    nsbuf.appendf(" %s", ns.c_str());
+                                binding.insert(scriptsNsInsertOffset, nsbuf);
+                            }
+                            appendEndTag(MANIFEST_TAG_SCRIPTS);
+                            appendEndCDATA();
+                            appendEndTag(MANIFEST_TAG_SCRIPTS);
+                            popNamespaces();
+                        }
+                        else
+                            appendEndTag(etag);
+
+                        level--;
+                        break;
+
+                    case XmlPullParser::END_DOCUMENT:
+                        throw makeStringExceptionV(ESDL_MANIFEST_ERR, "Unexpected end of document at line #%d, column #%d", xpp->getLineNumber(), xpp->getColumnNumber());
```

Comment:
Will we see an unexpected end of document, or will the parser through an exception? The parser definitely throughs exceptions in some situations, and I'm wondering if this is one of them.

Reply:
Tried a few different ways of 'unexpectedly' ending the document and in each case the parser or tokenizer threw an exception before it got to my case. I'll remove these checks.